inspire_json_merger/postprocess.py

```diff
@@ -0,0 +1,124 @@
+
+
+def remove_ordering_from_authors_conflicts(conflicts):
+    """Sorts authors and cleans up ordering information from conflicts."""
+    new_conflicts = []
+    for conflict in conflicts:
+        if '/authors' in conflict["path"]:
+            del conflict["value"][ORDER_KEY]
+        new_conflicts.append(conflict)
+    return new_conflicts
+
+
+def remove_ordering_from_authors_merged(merged):
+    """Sorts authors and cleans up ordering information from merged record."""
+    if 'authors' in merged:
+        merged["authors"].sort(key=itemgetter(ORDER_KEY))
+        for author in merged.get("authors"):
+            del author[ORDER_KEY]
+    return merged
+
+
+def postprocess_conflicts(conflicts, merged):
+    """Postprocessing conflicts to display only useful conflicts.
+
+    Before flattening and serializing to JSON patch, MERGE conflict looks like this:
+    ('MANUAL_MERGE', ('authors',), ( pmap(<ROOT>), pmap(<HEAD>), pmap(<UPDATE>) )),
+    This function takes out all MANUAL_MERGE and do as follows:
+        - Ignore ROOT
+        - Update merged result with HEAD from conflict by inserting the author
+            from HEAD at the right position.
+        - Look for ADD_BACK_TO_HEAD conflict with identical content as HEAD
+            change and remove it as it's already added to HEAD
+    Args:
+        conflicts(list): List of all possible conflicts
+        merged(dict): Merged document
+
+    Returns: A tuple containing the resulted merged record in json format and a
+        an list containing all generated conflicts.
+    """
+    new_conflicts = []
+    possible_dplicates = []
+    for conflict in conflicts:
+        if conflict[0] == "MANUAL_MERGE" and "authors" in conflict[1]:
+            new_conflict, merged, head = _process_author_manual_merge_conflict(conflict, merged)
+            new_conflicts.append(new_conflict)
+            possible_dplicates.append(head)
+        else:
+            new_conflicts.append(conflict)
+    new_conflicts = _remove_duplicates(new_conflicts, possible_dplicates)
+    return new_conflicts, merged
+
+
+def _remove_duplicates(conflicts, possible_duplicates):
+    new_conflicts = []
+    for conflict in conflicts:
+        if conflict[0] == "ADD_BACK_TO_HEAD" and "authors" in conflict[1] and conflict[2] in possible_duplicates:
+            continue
+        else:
+            new_conflicts.append(conflict)
+    return new_conflicts
+
+
+def _process_author_manual_merge_conflict(conflict, merged):
+    """Process author `MANUAL_MERGE` conflict
+
+    Conflict object is an tuple containing
+    (conflict_type, conflict_location, conflict_data)
+    where `conflict_data` is a tuple of: (ROOT, HEAD, UPDATE)
+    """
+    head = conflict[2][1]
+    position = head[ORDER_KEY]
+    if head not in merged["authors"]:
+        merged["authors"].append(dict(head))
+    new_conflict = Conflict("SET_FIELD", ("authors/{position}".format(position=position),), conflict[2][2])
```

Comment:
inserting at `position` is too naive, it won't work if other authors have been added/removed by the update. I think you need to insert at a position such that all elements coming before have a lower `ORDER_KEY` (or none) and all elements after have a higher `ORDER_KEY` (or none).

Reply:
Problem is that this is not an insert, it's an replace. And to replace field we have to replace correct one (and existing one). This looks to be safest way to determine field which should be replaced.